### ISSUE TITLE
Rule proposal: `prefer-query-selector`

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#171]
Prefer `querySelector` over `getElementById`, `querySelectorAll` over `getElementsByClassName` and `getElementsByTagName`. They are not faster than `querySelector` and it's better to be consistent.

## Fail

```js
document.getElementById('foo');
```

## Pass

```js
document.querySelector('#foo');
```

**Comment by SamVerschueren** [comment]
I agree, consistency all the way!

**Comment by Komorebi-Girl** [comment]
Hi, just checking: Is this issue still up for grabs? If so, I'd like to take it.

**Comment by Komorebi-Girl** [comment]
Okay, just for clarification: Do I need to change all of the getElementByIds, getElementsByClassNames, getElementsByTagNames and querySelectorAlls to querySelectors in a particular file? I'm still quite new to contributing to open source projects so thank you in advance for your help.

**Comment by sindresorhus** [comment]
@Komorebi-Girl I think you've misunderstood what this issue is about. This is not about replacing `getElementByIds` etc in this project, but rather create an ESLint linter rule that will warn about `getElementByIds`/etc in projects.

**Comment by Komorebi-Girl** [comment]
Ah, I see. Yes, I completely misunderstood. Well, in the case, I should leave this issue for someone else more equip while I try to get a better understanding of the project. Thank you, @sindresorhus, for your clarification!

**Comment by zoenkatz** [comment]
@sindresorhus, Can I take it? Thanks

**Comment by dilanSachi** [comment]
is this still open?
can i work on this?

**Comment by sindresorhus** [comment]
Sorry, missed this. Yes, this issue is still available.

**Comment by zoenkatz** [comment]
Ok cool, then I'm starting to work on it.

**Comment by yakov116** [comment]
@sindresorhus  is this what you had in mind?
I think the rule should only apply if the selector is a sting not a variable 
```javascript 
		if (document.getElementById(setting).type === 'checkbox') {
			document.getElementById(setting).checked = value;
```
Vs

``` javascript
		if (document.querySelector(`#${setting}`).type === 'checkbox') {
			document.querySelector(`#${setting}`).checked = value;
		} 
```

**Comment by yakov116** [comment]
@sindresorhus ?

**Comment by JounQin** [comment]
@sindresorhus Can we have an option for it? `slightly longer` is such redundant like ``#${setting}``.

**Comment by sindresorhus** [comment]
@JounQin No, just disable the rule if you don't want it.

**Comment by JounQin** [comment]
![image](https://user-images.githubusercontent.com/8336744/133808007-0bccd784-d690-4b7a-82ac-9c589ff3abd7.png)

![image](https://user-images.githubusercontent.com/8336744/133808256-e89fc0a6-0730-4991-b53a-6818fe9d7145.png)

`querySelector` can not always replace `getElementById` easily.

**Comment by sindresorhus** [comment]
There is no way to detect this, so I think this is a good case for just using an ESLint ignore comment.

**Comment by fisker** [comment]
[`CSS.escape()`](https://developer.mozilla.org/en-US/docs/Web/API/CSS/escape)?

**Comment by fisker** [comment]
![image](https://user-images.githubusercontent.com/172584/134495733-55a96dcc-a4c9-4c84-8984-d74b04bcb286.png)
